File: CPP/No168.cc
/**
 * Created by Xiaozhong on 2020/8/16.
 * Copyright (c) 2020/8/16 Xiaozhong. All rights reserved.
 */

#include "string"
#include "algorithm"

using namespace std;

class Solution {
public:
    string convertToTitle(int n) {
        string ans = "";
        while (n != 0) {
            n--;
            ans.push_back('A' + (n % 26));
            n /= 26;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: Java/src/No168.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

public class No168 {
    public String convertToTitle(int columnNumber) {
        StringBuilder builder = new StringBuilder();
        while (columnNumber > 0) {
            columnNumber--;
            builder.append((char) (columnNumber % 26 + 'A'));
            columnNumber /= 26;
        }
        return builder.reverse().toString();
    }

    public static void main(String[] args) {
        No168 no168 = new No168();
        System.out.println(no168.convertToTitle(2147483647));

    }
}
